applications/main/nfc/plugins/supported_cards/washcity.c

```diff
@@ -0,0 +1,200 @@
+
+    MfClassicData* data = mf_classic_alloc();
+    nfc_device_copy_data(device, NfcProtocolMfClassic, data);
+
+    do {
+        MfClassicType type = MfClassicTypeMini;
+        MfClassicError error = mf_classic_poller_sync_detect_type(nfc, &type);
+        if(error != MfClassicErrorNone) break;
+
+        data->type = type;
+        if(type != MfClassicType1k) break;
+
+        MfClassicDeviceKeys keys = {
+            .key_a_mask = 0,
+            .key_b_mask = 0,
+        };
+        for(size_t i = 0; i < mf_classic_get_total_sectors_num(data->type); i++) {
+            nfc_util_num2bytes(washcity_1k_keys[i].a, sizeof(MfClassicKey), keys.key_a[i].data);
+            FURI_BIT_SET(keys.key_a_mask, i);
+            nfc_util_num2bytes(washcity_1k_keys[i].b, sizeof(MfClassicKey), keys.key_b[i].data);
+            FURI_BIT_SET(keys.key_b_mask, i);
+        }
+
+        error = mf_classic_poller_sync_read(nfc, &keys, data);
+        if(error != MfClassicErrorNone) {
+            FURI_LOG_W(TAG, "Failed to read data");
+            break;
+        }
+
+        nfc_device_set_data(device, NfcProtocolMfClassic, data);
+
+        is_read = true;
+    } while(false);
+
+    mf_classic_free(data);
+
+    return is_read;
+}
+
+static bool washcity_parse(const NfcDevice* device, FuriString* parsed_data) {
+    furi_assert(device);
+
+    const MfClassicData* data = nfc_device_get_data(device, NfcProtocolMfClassic);
+
+    bool parsed = false;
+
+    do {
+        // Verify key
+        const uint8_t ticket_sector_number = 1;
+        const uint8_t ticket_block_number = 0;
+
+        const MfClassicSectorTrailer* sec_tr =
+            mf_classic_get_sector_trailer_by_sector(data, ticket_sector_number);
+
+        const uint64_t key = nfc_util_bytes2num(sec_tr->key_a.data, COUNT_OF(sec_tr->key_a.data));
+        if(key != washcity_1k_keys[ticket_sector_number].a) break;
+
+        // Parse data
+        const uint8_t start_block_num =
+            mf_classic_get_first_block_num_of_sector(ticket_sector_number);
+
+        const uint8_t* block_start_ptr =
+            &data->block[start_block_num + ticket_block_number].data[0];
+
+        uint32_t balance = nfc_util_bytes2num(block_start_ptr+2, 2);
+
+        uint32_t balance_eur = balance / 100;
+        uint8_t balance_cents = balance % 100;
+
+        size_t uid_len = 0;
+        const uint8_t* uid = mf_classic_get_uid(data, &uid_len);
+        
+        // Card Number is printed in HEX (equal to UID)
+        char card_number[2 * uid_len + 1];
+
+        for(size_t i = 0; i < uid_len; ++i) {
+            card_number[2 * i] = "0123456789ABCDEF"[uid[i] >> 4];
+            card_number[2 * i + 1] = "0123456789ABCDEF"[uid[i] & 0xF];
+        }
+
+        card_number[2 * uid_len] = '\0';
+
+        furi_string_printf(
+            parsed_data,
+            "\e#WashCity\nCard number: %s\nBalance: %lu.%02u EUR",
+            card_number,
+            balance_eur,
+            balancecentsi);
```

Comment:
Did you mean balance_cents?

Reply:
Yes, fixed that and clang format also. Thanks for bringing it up, probably a typo after build since it was running fine.

Reply:
FYI, merged on OFW.